source/core/StringUtilities.cpp

```diff
@@ -902,14 +916,23 @@ Boolean SubString::ParseDouble(Double *pValue, Boolean suppressInfNaN /*= false*
             if (length >= 9 && ComparePrefixIgnoreCase(Utf8Ptr("inity"), 5))
                 _begin += 5;
         }
+        if (bParsed && suppressInfNaN) {
```

Comment:
Nope.  Expand the elided code.  bParsed can be set to true again between the outer if and this test.